### DISCUSSION TITLE
Default multi-sorted columns bug - last column sorting acts differently (v8)

### DISCUSSION THREAD

**Discussion by zavtraleto** [TanStack/table#4902] [General]
Hello, I've attempted to implement default multi-sorting for table columns but encountered an issue. I've created a code sandbox based on the example of sorting, which you can find here: https://codesandbox.io/p/sandbox/tanstack-sorting-bug-60owh6?file=%2Fsrc%2Fmain.tsx%3A91%2C1

As you'll see, we have default sorting for multiple columns, and when you click on the first one, it turns off sorting for all other columns. However, there's a problem with the last one - it doesn't turn off sorting for other columns.

In my example, if you click on the 'firstName' column, it functions correctly. However, the 'lastName' column doesn't operate as expected.

Is it an issue or a logical behavior?

**Comment by ahussein3** [comment]
@tannerlinsley can I have an explanation for that? is this a bug or it's made like this by design? And if there is a way to overcome it ?
